hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/SqlAnalyzeStatement.java

```diff
@@ -80,4 +82,64 @@ public void unparse(final SqlWriter writer, final int leftPrec, final int rightP
 
         query.unparse(writer, leftPrec, rightPrec);
     }
+
+    public void validate(SqlValidator validator) {
+        Set<String> optionNames = new HashSet<>();
+        jobConfig.setMetricsEnabled(true);
+        jobConfig.setStoreMetricsAfterJobCompletion(true);
+
+        jobConfig.setSplitBrainProtection(false);
+        jobConfig.setAutoScaling(false);
+        jobConfig.setSuspendOnFailure(false);
+
+        for (SqlNode option0 : options) {
+            SqlOption option = (SqlOption) option0;
+            String key = option.keyString();
+            String value = option.valueString();
+
+            if (!optionNames.add(key)) {
+                throw validator.newValidationError(option, RESOURCE.duplicateOption(key));
+            }
+
+            switch (key) {
+                case "processingGuarantee":
+                    switch (value) {
+                        case "exactlyOnce":
+                            jobConfig.setProcessingGuarantee(EXACTLY_ONCE);
+                            break;
+                        case "atLeastOnce":
+                            jobConfig.setProcessingGuarantee(AT_LEAST_ONCE);
+                            break;
+                        case "none":
+                            jobConfig.setProcessingGuarantee(NONE);
+                            break;
+                        default:
+                            throw validator.newValidationError(option.value(),
+                                    RESOURCE.processingGuaranteeBadValue(key, value));
+                    }
```

Comment:
This part has a duplicate in `SqlCreateJob.validate()`. It should be moved to a utility method: `ProcessingGuarantee parseProcessingGuarantee(SqlValidator validator, SqlOption option)` in `com.hazelcast.jet.sql.impl.parse.ParseUtil`.

Reply:
I don't see any value to do it. Here we have a straightforward and understandable algorithm. Moving it in some utility method just smears the logic.

Reply:
Parsing of processing guarantee should be centralized. If a new option is introduced or an existing one is removed in the future, we will need to find all places where processing guarantee is parsed and do the same modification to each of them.

Reply:
> **If** a new option is introduced or an existing one is removed

This is a key word. Personally, I prefer readability here, and I can sacrifice 'clean code' recommendations or 'finding duplicates'.

Reply:
>Here we have a straightforward and understandable algorithm
... in 2 copies which hopefully are in sync and will be in sync in the future

@burakgok talks only about `ProcessingGuarantee` and I agree - DRY. The entire switch for options is slightly different because different options are supported. Moving that to a more generic method gives less benefit currently.

Reply:
How `ProcessingGuarantee` is parsed and how it is used are separate concerns. The utility is just for parsing `ProcessingGuarantee`, so it shouldn't expect `JobConfig`. `key` and `value` are redundant since they can be obtained from the `option`.